qmlui/qml/KeyPad.qml

```diff
@@ -95,18 +101,60 @@ Rectangle
                 {
                     tapTimer.stop()
                     tapButton.border.color = UISettings.bgMedium
-                    tapTimeValue = 0
+                    lastTap = 0
+                    tapHistory = []
                 }
                 else
                 {
                     var currTime = new Date().getTime()
-                    if (tapTimeValue != 0)
+                    
+                    if (lastTap != 0 && currTime - lastTap < 1500)
                     {
-                        keyPadRoot.tapTimeChanged(currTime - tapTimeValue)
-                        tapTimer.interval = currTime - tapTimeValue
+                        var newTime = currTime - lastTap
+                        
+                        tapHistory.push(newTime)
+                        var tapHistorySorted = []
+
+                        //reduce size to only 16 taps
+                        while (tapHistory.length > 16) tapHistory.splice(0,1)
+
+                        //copy tap history to sort it
+                        for(var i = 0; i < tapHistory.length; i++)
```

Comment:
space after `for`
But how about this?
`tapHistorySorted = tapHistory.slice();`

Reply:
Adjusted!
Yes, this is far better.